models/classes/user/implementation/UserSettingsService.php

```diff
@@ -65,10 +75,29 @@ public function get(core_kernel_classes_Resource $user): UserSettingsInterface
             $timezone = (string) current($props[GenerisRdf::PROPERTY_USER_TIMEZONE]);
         }
 
-        return new UserSettings(
+        $userSettings = new UserSettings(
             $timezone ?? $this->defaultTimeZone,
             $uiLanguageCode ?? null,
             $dataLanguageCode ?? null
         );
+
+        if (!empty($props[GenerisRdf::PROPERTY_USER_INTERFACE_MODE])) {
+            $userSettings->setSetting(
+                UserSettingsInterface::INTERFACE_MODE,
+                current($props[GenerisRdf::PROPERTY_USER_INTERFACE_MODE])->getUri()
+            );
+        }
+
+        return $userSettings;
+    }
+
+    public function getCurrentUserSettings(): UserSettingsInterface
+    {
+        $currentUser = $this->userService->getCurrentUser();
+        if ($currentUser) {
+            return $this->get($currentUser);
+        }
+
+        return new UserSettings($this->defaultTimeZone);
```

Comment:
Please, never glue `if` with other `ifs` or other instructions.

```suggestion
        $currentUser = $this->userService->getCurrentUser();
        
        if ($currentUser) {
            return $this->get($currentUser);
        }

        return new UserSettings($this->defaultTimeZone);
```